### ISSUE TITLE
Vertical nav has inaccessible button names

### ISSUE THREAD

**Issue by gnomeontherun** [vmware/clarity#3574]
## Describe the bug

The Vertical Nav button in the Collapse & Expand Navigation example lacks an accessible name because it is empty.

The Nav trigger button lacks accessible name. Screen reader users will not know the purpose of this button when they read it. 

## How to reproduce

https://v1.clarity.design/vertical-nav/basic-structure/charmander

- Using a screen reader go to https://v1.clarity.design/vertical-nav/basic-structure/charmander
- Find the Collapse & Expand Navigation examples 
- Activate the trigger button for the first example 
- Press the Tab key once and notice that you do not hear the purpose of the button you are reading
- Read the nav-trigger button and observe that the screen reader does not announce any information about the purpose of this button.

## Expected behavior

Provide accessible names for these buttons.

## Versions

**App**

* Angular: [e.g. 6]
* Node: [e.g. 8.10.0]
* Clarity: [e.g. 0.12.5]

**Device:**

* Type: [e.g. MacBook]
* OS: [e.g. iOS]
* Browser [e.g. chrome, safari]
* Version [e.g. 22]

## Additional notes

_Add any other notes about the problem here._

**Comment by andreiconstantinbytex** [comment]
I've investigated this and it seems that a simple "aria-label" applied to all of the sidenav buttons should suffice. I would like to be assigned to this issue if you consider the aforementioned solution is the best one. Thanks

**Comment by Jinnie** [comment]
I believe @Shijir is already working on this. Please wait for an update from him. And thanks for your research and commitment :)

**Comment by gnomeontherun** [comment]
@andreiconstantinbytex This issue and https://github.com/vmware/clarity/issues/3576 are both open. Shijir's work was focused on the focus management only. These could be tackled together, if you'd like to take it up I will assign you!

**Comment by andreiconstantinbytex** [comment]
Can you please assign me on #3576 and #3574?

**Comment by andreiconstantinbytex** [comment]
Also, what is the status of #3575 ?

**Comment by andreiconstantinbytex** [comment]
I have 2 ways of tackling this issue:
1. Treat it as a toggle-able button which has aria-label set(e.g.'Toggle') and aria-pressed= true | false, depending on whether the menu is opened or not
2. Treat the button as a menu button which has aria-haspopup="menu" and two types of aria-label, either 'Expand' or 'Collapse', depending on whether the menu is opened or not

These changes would be made on the vertical-nav component itself.

Would these be the best approaches or do you have anything to add?

**Comment by gnomeontherun** [comment]
The button should get an `aria-label` since its just an icon only button, and I believe `aria-expanded` is better to explain the expanded state of the nav. The toggle button isn't really a menu, so I don't think we want to go that direction.

**Comment by andreiconstantinbytex** [comment]
Just as a side-note: this issue also occurs on v2.

**Comment by gnomeontherun** [comment]
This should be closed by #3860

**Comment by bdryanovski** [comment]
Checked and seems to be covered by #3860 so I'm closing it.

**Comment by chlane** [comment]
Fix verified on https://clarity.design/documentation/vertical-nav/collapsible-nav/normal
